### PR TITLE
Added namespace Filtering to GetManagedLiveObjs

### PR DESCRIPTION
**Pull request by shubhamdixit863**
Fixes  https://github.com/numaproj/numaplane/issues/8

### Modifications

Modified `GetManagedLiveObjs` method for `LiveStateCache` cluster caching to filter out object based on name and namespace both.

### Verification

Tested in local cluster, by applying file `numaplane_controller_rollout.yaml`, it does the proper logging mentioned in the code

### CODE REVIEW

**Comment by juliev0** [comment]
looks good but can you test it by using [this](https://github.com/numaproj/numaplane/blob/main/config/samples/numaplane.numaproj.io_v1alpha1_numaflowcontrollerrollout.yaml) sample NumaflowControllerRollout? (Maybe add a temporary log line to confirm the code is being executed)

**Comment by shubhamdixit863** [comment]
> looks good but can you test it by using [this](https://github.com/numaproj/numaplane/blob/main/config/samples/numaplane.numaproj.io_v1alpha1_numaflowcontrollerrollout.yaml) sample NumaflowControllerRollout? (Maybe add a temporary log line to confirm the code is being executed)

Yeah  i tested it ,works fine

**Review by juliev0 — APPROVED** [review]
LGTM but let's have @xdevxy sign off since it's related to code she wrote...

File: internal/sync/cache.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ type LiveStateCache interface {
 	// GetClusterCache returns synced cluster cache
 	GetClusterCache() (clustercache.ClusterCache, error)
 	// GetManagedLiveObjs returns state of live objects which correspond to target objects with the specified ResourceInfo name.
```
Comment (xdevxy) [inline internal/sync/cache.go:78]: nit: update the comment accordingly.

**Review by xdevxy — APPROVED** [review]

**Pull request merged**